src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -91,6 +93,39 @@ - (id)strongObject
 }
 @end
 
+// Stores essential-for-logging attributes immutably for use in logs
+@interface MTRDeviceEssentialAttributes : NSObject
+@property (readonly) UInt16 vendorID;
```

Comment:
Why not allow these to be mutable? Also I forget, is atomic, assigned present by default? Note... we also should be using NSUIntegers, or NSNumber*

Reply:
(Preferably NSNumber *, to follow other stuff)

Reply:
`atomic` is default, i think `assign` is default as well.  `NSNumber *` sounds fine.

Reply:
As far as mutability:  I was hoping to set this up as a set of attributes that are unchanging throughout the `MTRDevice`'s lifetime and therefore only have to be pulled from their respective clusters _once_ at a predictable point in the lifecycle.  Another valid option is to make the properties computed and pull them (from cache?) every time we want to log them.

I don't have a comprehensive picture of how often we want to use these properties in logs or how often those logs will be called.  I also am not sure about all scenarios where our attribute caches get cleared. I erred on the conservative side here, but that could be premature optimization on my part.  Thoughts?

Reply:
(Discussed earlier today - probably will want to frequently log some changeable properties soon, so immutability isn't going to work out in the long run.)